lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,376 @@
+		}
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether a given block type should be visible.
+	 *
+	 * @return WP_Error|bool True if the block type is visible, otherwise false.
+	 */
+	protected function check_read_permission() {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error( 'rest_cannot_view', __( 'Sorry, you are not allowed to manage block types.', 'gutenberg' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the block, if the name is valid.
+	 *
+	 * @param string $name Block name.
+	 * @return WP_Block_Type|WP_Error Block type object if name is valid, WP_Error otherwise.
+	 */
+	protected function get_block( $name ) {
+		$block_type = $this->registry->get_registered( $name );
+		if ( empty( $block_type ) ) {
+			return new WP_Error( 'rest_block_type_invalid', __( 'Invalid block type.', 'gutenberg' ), array( 'status' => 404 ) );
+		}
+
+		return $block_type;
+	}
+
+	/**
+	 * Retrieves a specific block type.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+		$data = $this->prepare_item_for_response( $block_type, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a block type object for serialization.
+	 *
+	 * @param WP_Block_Type   $block_type block type data.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 *
+	 * @return WP_REST_Response block type data.
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( in_array( 'attributes', $fields, true ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( in_array( 'is_dynamic', $fields, true ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$schema       = $this->get_item_schema();
+		$extra_fields = array(
+			'name',
+			'category',
+			'editor_script',
+			'script',
+			'editor_style',
+			'style',
+		);
+		foreach ( $extra_fields as $extra_field ) {
+			if ( in_array( $extra_field, $fields, true ) ) {
+				$field                = ( isset( $block_type->$extra_field ) ) ? $block_type->$extra_field : false;
+				$data[ $extra_field ] = rest_sanitize_value_from_schema( $field, $schema['properties'][ $extra_field ] );
```

Comment:
We can skip this sanitization. We don't apply it in the post types controller for instance.

Reply:
I don't like that in core. We can do better here.

Reply:
What benefit does that get us? If an invalid value is set in PHP, it'll break anything in PHP that reads from it. If we want to validate those properties they should be validated when the block type is registered.

Reply:
These are class properties and are not typed. What if one of the fields are set to an array or object. We should be type checking all output.

Reply:
We should check that type on registration. If somewhat sets an `array` for the script for instance, then the server side code and logic will be entirely borked.